Doc/tools/extensions/pyspecific.py

```diff
@@ -22,6 +22,7 @@
 
 from sphinx import addnodes
 from sphinx.builders import Builder
+from sphinx.errors import NoUri
```

Comment:
Looks like this is not available on the version of Sphinx that's used in CI. I'm pretty sure it's pinned somewhere as well.

Maybe we should just warn on all exceptions and skip the reference?

Reply:
It moved recently, we could try both imports.